app/js/lib/intent-parser.js

```diff
@@ -17,80 +18,106 @@ Remind Guillaume on Thursdays evening to go to his drawing class.
-      `Remind [user] to [action] on [time].`,
-      `Remind [user] to [action] by [time].`,
-      `Remind [user] at [time] to [action].`,
-      `Remind [user] on [time] to [action].`,
-      `Remind [user] by [time] to [action].`,
-      `Remind [user] that it is [action] on [time].`,
-      `Remind [user] that [time] is [action].`,
+      `Remind [users] to [action] at [time].`,
+      `Remind [users] to [action] on [time].`,
+      `Remind [users] to [action] by [time].`,
+      `Remind [users] at [time] to [action].`,
+      `Remind [users] on [time] to [action].`,
+      `Remind [users] by [time] to [action].`,
+      `Remind [users] that it is [action] on [time].`,
+      `Remind [users] that it is [action] at [time].`,
+      `Remind [users] that it is [action] by [time].`,
+      `Remind [users] that [time] is [action].`,
+      `Remind [users] that [action] at [time].`,
+      `Remind [users] that [action] on [time].`,
+      `Remind [users] that [action] by [time].`,
     ],
     placeholders: {
-      user: '( \\S+ | \\S+,? and \\S+ )',
+      users: '( \\S+ | \\S+,? and \\S+ )',
       action: '(.+)',
       time: '(.+)',
     },
     // @see http://www.unicode.org/cldr/charts/29/summary/en.html#4
     punctuation: new RegExp(
       `[-‐–—,;:!?.…'‘’"“”()\\[\\]§@*/&#†‡′″]+$`, 'u'),
+    // @see http://www.unicode.org/cldr/charts/29/summary/en.html#6402
+    listBreaker: new RegExp(`,|, and\\b|\\band\\b`, 'gu'),
   },
   fr: {
     patterns: [
-      `Rappelle [user] de [action] [time].`,
-      `Rappelle [user] d'[action] [time].`,
-      `Rappelle-[user] de [action] [time].`,
-      `Rappelle-[user] d'[action] [time].`,
+      `Rappelle [users] de [action] [time].`,
+      `Rappelle [users] d'[action] [time].`,
+      `Rappelle-[users] de [action] [time].`,
+      `Rappelle-[users] d'[action] [time].`,
     ],
     placeholders: {
-      user: '( \\S+ | \\S+ et \\S+ )',
+      users: '( \\S+ | \\S+ et \\S+ )',
       action: '(.+)',
       time: '(.+)',
     },
     punctuation: new RegExp(
       `[-‐–—,;:!?.…’"“”«»()\\[\\]§@*/&#†‡]+$`, 'u'),
+    listBreaker: new RegExp(`,|\\bet\\b`, 'gu'),
   },
   ja: {
     patterns: [
-      `[time][action]を[user]に思い出させて。`,
-      `[time][user]に[action]を思い出させて。`,
-      `[time][user]は[action]と言うリマインダーを作成して。`,
+      `[time][action]を[users]に思い出させて。`,
+      `[time][users]に[action]を思い出させて。`,
+      `[time][users]は[action]と言うリマインダーを作成して。`,
     ],
     placeholders: {
-      user: '(\\S+|\\S+と\\S+)',
+      users: '(\\S+|\\S+、\\S+)',
       action: '(.+)',
       time: '(.+)',
     },
     punctuation: new RegExp(
       `[-‾_＿－‐—―〜・･,，、､;；:：!！?？.．‥…。｡＇‘’"＂“”(（)）\\[［\\]］{｛}｝` +
       `〈〉《》「｢」｣『』【】〔〕‖§¶@＠*＊/／\＼&＆#＃%％‰†‡′″〃※]+$`, 'u'),
+    listBreaker: new RegExp(`、`, 'gu'),
   },
 };
 
 export default class IntentParser {
   constructor(locale = 'en') {
     this.locale = locale;
-    this[p.regexps] = {};
+    this[p.confirmation] = new Confirmation(locale);
```

Comment:
`Confirmation` seems to be more of a formatting thing rather than a parser though, I can't think of a good name off the top of my head either.  Some kind of class to encapsulate the two would be good.